### DISCUSSION TITLE
How do I show user is typing?

### DISCUSSION THREAD

**Discussion by azambuilds** [flyerhq/flutter_chat_ui#583] [Q&A] [answered]
I saw there was a discussion about it but I can't find the documentation or it in example. I am building a chatbot and I need to show the user that the chatbot is loading (with a "chatbot is typing" indicator).

Thanks!

**Comment by demchenkoalex** [comment]
there is `typingIndicatorOptions` class, and it has `typingUsers` array - you can pass id of your bot there and it will show. Loading can be done outside of the package or via some overlay/popup?

**Reply by azambuilds** [reply]
Thanks I completely missed the typingIndicatorOptions field for the Chat widget. Thanks again!

**Comment by demchenkoalex** [comment, marked as answer]
for v2 pointing to this (as of right now not implemented, but if discussion will be answered means it is implemented) https://github.com/flyerhq/flutter_chat_ui/discussions/191

**Reply by qasim90** [reply]
@demchenkoalex Is typing indicator supported in v2 yet?

**Reply by demchenkoalex** [reply]
it is, there is an example in `local.dart`
